### ISSUE TITLE
Having issues handling 401 from server

### ISSUE THREAD

**Issue by martosoler** [websanova/vue-auth#214]
Hi,

After my server returns a 401 the `_invalidToken` function gets called as expected:

```
_invalidToken: function (res) {
    if (res.status === 401) {
      console.log('invalid token');
      this.options.logoutProcess.call(this, res, {redirect: this.options.authRedirect});
    }
  },
```
I haven't customized the `logoutProcess` function so the first line is:
```
__cookie.delete.call(this);
```
I have put a log inside that `delete` function:
```
delete: function(name) {
            setCookie.call(this, name, '', -12096e5);
        }
```
and the `name` parameter is `undefined`. 
In my case, I'm storing the token in a cookie named `token` and have specified in the options that the `tokenName` is 'token'.
Is this a bug or how can I get the cookie removed when the logout is called internally?

Another question, I have seen in the token.js file that the `get` function favors local storage over cookies, is there a way to change that?
```
get: function (name) {
            name = tokenName.call(this, name);

            if (isLocalStorageSupported()) {
                return localStorage.getItem(name);
            }
            else {
                return __cookie.get.call(this, name);
            }
        },
```
As I said before, in our application we don't use the Local storage.

I'm using the following versions:

- vue-auth (2.12.1-beta)
- vue (2.3.3)
- vue-router (2.3.1)
- vue-axios (2.0.2)

Thanks

**Comment by websanova** [comment]
Hey,

Yes, the cookie thing is a bug, should be using the name from options.

As for the local storage, cookies, I can add another option for that,
probably as an array so the storage order and availability can be set.

Should be able to get this in by end of weekend.

**Comment by websanova** [comment]
Hey, so the cookie thing should be fine actually. Note that cookies only get expired not actually deleted so make sure you check the dates.

Otherwise I've added a  new option `tokenStore` where you can set storage preferences.

Make sure to grab the latest (`v2.16.0-beta`).

https://github.com/websanova/vue-auth/blob/master/docs/Options.md#tokenstore-localstorage-cookie

**Comment by martosoler** [comment]
Thanks for the changes, I'm using the `tokenStore` property so no more LocalStorage, that's great !!

I'm a bit confused now, in our app when the customer authenticates successfully, a set-cookie header is returned with the JWT token. That cookie is called _token_, that's why I put _token_ to the `tokenName` property but now I see two cookies: `default_token` and `token`.

I have implemented the `check` method by _checking_ the existence of the cookie the server returns, is this ok?

Regarding your statement about not being able to delete cookies, you are actually deleting the _rememberMe_ cookie and I'm deleting the cookie when the user clicks Logout. I wish the cookie to be also deleted when the `_logoutProcess` gets called by a 401 from the server. Do you think that's possible or even correct?

Thanks

**Comment by websanova** [comment]
Yes, the plugin will prepend it because there can be multiple tokens stored
when impersonating other users.

Can you set it as default_token from the server?

**Comment by martosoler** [comment]
Unfortunately I do not completely own the server so I found that very difficult to change. 

Several other (Angular) apps are following the same process with the cookie named _token_ and this one is our first Vue app. 

Where is the impersonation logic you mention? Is there a way to disable it?

**Comment by websanova** [comment]
Well try impersonating logic has nothing to do with how the cookies are
named.

I will just add some options for the prefixes so that they can be nulled
out.

Give me a day to push that.

On Aug 7, 2017 8:35 PM, "Leandro Martin Soler" <notifications@github.com>
wrote:

Unfortunately I do not completely own the server so I found that very
difficult to change.

Several other (Angular) apps are following the same process with the cookie
named *token* and this one is our first Vue app.

Where is the impersonation logic you mention? Is there a way to disable it?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/websanova/vue-auth/issues/214#issuecomment-320651547>,
or mute the thread
